app/src/main/java/com/mmm/parq/fragments/DriverHomeFragment.java

```diff
@@ -66,48 +79,127 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         SupportMapFragment mapFragment = (SupportMapFragment) this.getChildFragmentManager()
                 .findFragmentById(R.id.map);
         mapFragment.getMapAsync(this);
+        mQueue = HttpClient.getInstance(getActivity().getApplicationContext()).getRequestQueue();
+        mGson = new Gson();
+        mRelativeLayout = (RelativeLayout) view.findViewById(R.id.driver_home_layout);
+        mCurrentSpot = null;
 
         mQueue = HttpClient.getInstance(getActivity().getApplicationContext()).getRequestQueue();
         mFindParkingButton = (Button) view.findViewById(R.id.findparkingbutton);
         mFindParkingButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                // Reserve a spot for the user.
-                requestReservation(new HttpClient.VolleyCallback<String>() {
+                // If the button is in 'find spot' mode, reserve a spot
+                if (getString(R.string.find_spot_button_text).equals(mFindParkingButton.getText())) {
+                    // Reserve a spot for the user.
+                    reserveSpot();
+                } else {
+                    // Otherwise, navigate to the spot
+                    // TODO(kenzshelley) actually send an intent to navigate to the currently reserved spot.
+                    if (mCurrentSpot == null) {
+                        Log.d(CLASS, "No spot has been reserved!");
+                    }
+                }
+            }
+        });
+
+        return view;
+    }
+
+    private void reserveSpot() {
+        requestReservation(new HttpClient.VolleyCallback<String>() {
+            @Override
+            public void onSuccess(String response) {
```

Comment:
So the basic problem is that we have a lot of requests that are dependent upon other requests, and as far as I know there isn't a pretty way to chain requests together in Java/Volley.

One approach is I think what you just suggested, which is to have one function that you call inside of the first 'onSuccess', and then inside of that function you make the next request, and so on. The problem with that is that it makes it hard to see what's going on consecutively; i.e. i want to see that I made a request to make a reservation, then I made a request to get directions to the spot reserved for me, and then I made a request to get spot information for this reservation. Having all of the requests buried inside of different functions seemed like it was going to make it harder to follow to me....but I'm not really sure. 

The nested callbacks are definitely kind of ugly too. What would be ideal is if we could do something similar similar to js promises where you can just say: 

```
requestReservation(blah).then(function(blah) {
  do stuff
  return requestThing();
}).then(function(meh) {
  do stuff;
  return requestThing2();
}).catch(function(error) {

})
```

So yeah I'm not really sure what the best approach is I guess